src/xradio/image/_util/_zarr/zarr_low_level.py

```diff
@@ -275,7 +288,23 @@ def create_data_variable_meta_data_on_disk(
                 chunks.append(xds_dims[d])
 
         # print(chunks,shape)
-        write_json_file(zattrs, os.path.join(data_variable_path, ".zattrs"))
+        # assuming data_variable_path has been set compatibly
+        zattrs_file = os.path.join(data_variable_path, ".zattrs")
+
+        if isinstance(fs, s3fs.core.S3FileSystem):
+            with fs.open(zattrs_file, "w") as file:
+                json.dump(
+                    zattrs,
+                    file,
+                    indent=4,
+                    sort_keys=True,
+                    ensure_ascii=True,
+                    separators=(",", ": "),
+                    cls=NumberEncoder,
+                )
+        else:
+            # default to assuming we can use primitives
+            write_json_file(zattrs, zattrs_file)
```

Comment:
The S3 conditional was simpler to add here than it would have been to update `write_json_file` because this function is where we have access to the (renamed) xradio._utils.zarr.common._get_file_system_and_items. It might make sense to refactor the latter into the former though.

Reply:
I agree that xradio._utils.zarr.common._get_file_system_and_items should be refactored. Maybe split into two functions get_file_system and get_items (I also think we need a more descriptive name).